### ISSUE TITLE
Inspection for inappropriate Type comparisons in Select Case Blocks

### ISSUE THREAD

**Issue by ThunderFrame** [rubberduck-vba/Rubberduck#3050]
The following block has a Boolean expression in the `Select Case` statement, but the `Case` statements are using non-Boolean types:

```vb
  Select Case VBA.Rnd > 0.5
    Case -5 'Implicitly True
      Stop
    Case -4 'Unreachable
      Stop
    Case Else
      Stop
  End Select
```

Rubberduck should have an inspection that identifies the type mismatches, and the implied casts that they entail.

In the case above, the fix could be either of the following (where my preference would be for the first):

```vb
  Select Case VBA.Rnd
    Case > 0.5
      Stop
    Case Else
      Stop
  End Select
```

or

```vb
  Select Case VBA.Rnd > 0.5
    Case True
      Stop
    Case Else
      Stop
  End Select
```

The Variant comparison bug #3047 can show up here too.

**Comment by ThunderFrame** [comment]
references #1584

**Comment by bclothier** [comment]
We can consider this closed by PR #3868 - additional issues with the inspection should be opened separately.
